src/pages/event.tsx

```diff
@@ -0,0 +1,157 @@
+import { useLocale } from "@/locales";
+import { useState, useRef } from "react";
+function SingleEvent({
+  bgColor,
+  borderColor,
+  toReverse,
+  textColor,
+  image,
+  eventInfo,
+  price,
+}: {
+  bgColor: string;
+  borderColor: string;
+  toReverse: boolean;
+  textColor: string;
+  image: string;
+  eventInfo: string[];
+  price: string;
+}) {
+  const t = useLocale();
+  const [modalState, setModal] = useState(false);
+  const modalRef = useRef();
+
+  const openModal = () => {
+    setModal(true);
+  };
+
+  const closeModal = () => {
+    setModal(false);
+  };
+
+  const handleOverlayClick = (event) => {
```

Comment:
Unfortunately typescript compiler big dumb dumb so the parameter has to be explicitly typed here (to `React.MouseEvent`)

Reply:
Nice, I kinda just ignored the compiler complaints.
Fixed now.